core/templates/dev/head/css/oppia.css

```diff
@@ -1189,10 +1189,50 @@ pre.oppia-pre-wrapped-text {
   position: relative;
 }
 
-.oppia-editable-section:hover .oppia-state-content-display,
-.oppia-editable-section:hover .oppia-rule-preview {
-  background: #eee;
+/* this class below is used to graymask
+ * outcome section when one hovers
+ * over the editable section
+ */
+
+.oppia-editable-section .oppia-rule-preview-section-mask {
+  background-color: #eee;
+  bottom: 0;
+  height: 100%;
+  opacity: 0;
+  position: absolute;
+  top: 0;
+  width: 100%;
+  z-index: 10;
+}
+
+.oppia-editable-section:hover .oppia-rule-preview-section-mask {
+  border-radius: 4px;
+  opacity: 0.4;
+  transition: all 200ms;
+  -webkit-transition: all 200ms;
+}
+
+/* this class below is used to graymask
+ * editable section when one hovers
+ * over the editable section
+ */
+
+.oppia-editable-section .oppia-editable-section-mask {
```

Comment:
Can you deduplicate this by combining it with the above? I.e. have a selector of the form

```
.oppia-editable-section .oppia-rule-preview-section-mask,
.oppia-editable-section .oppia-editable-section-mask {
  ...
```